### PR TITLE
multi-miner / single LID - piece doctor

### PR DESCRIPTION
**Pull request by dirkmc**
TODO:
- [x] Test migration on local devnet
- [ ] Test in production

### CODE REVIEW

File: extern/boostd-data/svc/setup_yugabyte_test_util.go [hunk 1]
```diff
@@ -15,37 +18,128 @@ var tlog = logging.Logger("ybtest")
...
-	ConnectString: "postgresql://postgres:postgres@localhost:5433",
+	ConnectString: "postgresql://postgres:postgres@localhost:5433?sslmode=disable",
 }
 
-func SetupYugabyte(t *testing.T) {
-	ctx := context.Background()
+func init() {
+	// Uncomment when testing against a local yugabyte instance.
+	//TestYugabyteSettings = TestYugabyteSettingsLocal
+}
```
Comment (nonsense) [inline extern/boostd-data/svc/setup_yugabyte_test_util.go:33]: We could probably just move this to `setup_yugabyte_test_util_test.go` which should be compiled only if you run `go test`, no?

File: piecedirectory/doctor.go [hunk 2]
```diff
@@ -135,10 +142,15 @@ func (d *Doctor) checkPiece(ctx context.Context, pieceCid cid.Cid, lu *sectorsta
 		}
 	}
 
+	if !hasDealsOnThisMiner {
+		doclog.Debugw("ignoring piece as it is not present in any deals on this miner", "piece", pieceCid.String(), "miner", d.maddr.String())
+		return nil
+	}
```
Comment (nonsense) [inline piecedirectory/doctor.go:148]: I'm confused -- we go over all pieces stored in LID on all boostd instances?

File: extern/boostd-data/svc/setup_yugabyte_test_util.go (hunk 1 above)
Comment (dirkmc) [reply extern/boostd-data/svc/setup_yugabyte_test_util.go:33]: We only want it when running `go test` on a local machine though (not in CI)

File: piecedirectory/doctor.go (hunk 2 above)
Comment (dirkmc) [reply piecedirectory/doctor.go:148]: This is just a sanity check. There are two different processes that populate the list of pieces to check, and that add / remove deals. So it's possible that
- the process that adds to the list of pieces to check adds the piece from deal A
- another process deletes deal A
- GetNextPiecesToCheck returns the piece cid from deal A
- there are no deals for that piece

File: extern/boostd-data/yugabyte/create.cql [hunk 3]
```diff
@@ -1,6 +1,4 @@
-create keyspace if not exists idx with replication = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };
```
[the remark this answers is not in the document]
Comment (dirkmc) [reply extern/boostd-data/yugabyte/create.cql:1]: That command is now run as a preparatory step in the CreateKeyspace function. I moved it there so we can create a different keyspace for tests.

File: piecedirectory/doctor.go (hunk 2 above)
Comment (dirkmc) [reply piecedirectory/doctor.go:148]: You're right that this should happen very rarely. I updated the log level to a warning.

**Review by nonsense — APPROVED** [review]

**Pull request merged**